### PR TITLE
Eslint plugin (against dev)

### PR DESCRIPTION
**Pull request by omerlh**
This template is for your information. Please clear everything when submitting your pull request.

**Note: DefectDojo is now on Python3.5 and Django 2.2.x Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [X] Your code is flake8 compliant.
- [X] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted).
- [X] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [X] Add applicable tests to the unit tests.
- [X] Add the proper label to categorize your PR.

Current accepted labels for PRs:
- Import Scans (for new scanners/importers)
- enhancement
- performance
- feature
- bugfix
- maintenance (a.k.a chores)
- dependencies
- New Migration

Sorry for the mess folks! This is instead of #2465 - moved to use eslint instead of devskim. ESLint has support for many security rules, so this is why I want to use it instead.

See additional PRs - docs: DefectDojo/Documentation#100 and samples: DefectDojo/sample-scan-files#54

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Thanks for the PR. Would you mind adding one more test? For an "empty" report, i.e. a report without findings. Sometimes we see parsers bark on those, breaking CI/CD builds, etc. So it would be helpful to  have a unit test for that.

**Comment by omerlh** [comment]
Sure, added

File: dojo/tools/eslint/parser.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+class ESLintParser(object):
+    def _convert_eslint_severity_to_dojo_severity(self, eslint_severity):
+        if eslint_severity == 2:
+            return "High"
+        elif eslint_severity == 1:
+            return "Medium"
+        else:
+            return None
```
Comment (Maffooch) [inline dojo/tools/eslint/parser.py:14]: Is there never an opportunity for any vulnerable to have a severity of critical or lower than medium?

File: dojo/tools/eslint/parser.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        for item in data:
+            categories = ''
+            language = ''
+            mitigation = ''
+            impact = ''
+            references = ''
+            findingdetail = ''
+            title = ''
+            group = ''
+            status = ''
```
Comment (Maffooch) [inline dojo/tools/eslint/parser.py:33]: Majority of these are unused. No point in keeping them around.

File: dojo/tools/eslint/parser.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                            test=test,
+                            active=False,
+                            verified=False,
+                            description=findingdetail,
+                            severity=sev.title(),
+                            numerical_severity=Finding.get_numerical_severity(sev),
+                            file_path=item["filePath"],
+                            line=message["line"],
+                            url='N/A',
+                            static_finding=True)
```
Comment (Maffooch) [inline dojo/tools/eslint/parser.py:57]: Findings have a few required fields that are not being acknowledged here such as mitigation and impact.

File: dojo/tools/eslint/parser.py (hunk 1 above)
Comment (omerlh) [reply dojo/tools/eslint/parser.py:14]: Yep, ESLint support only error or warning.

File: dojo/tools/eslint/parser.py (hunk 3 above)
Comment (omerlh) [reply dojo/tools/eslint/parser.py:57]: fixed

File: dojo/tools/eslint/parser.py (hunk 2 above)
Comment (omerlh) [reply dojo/tools/eslint/parser.py]: fixed
